.github/workflows/unit-tests.yml

```
@@ -0,0 +1,49 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2020 - 2024 Pionix GmbH and Contributors to EVerest
+
+name: Execute Vitest Unit Tests
+
+on:
+  pull_request:
+    branches:
+      - main
+      - spr/main/*
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
+  workflow_dispatch: {}
+
+jobs:
+  package:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
```

`.nvmrc` specifies to use node 20. We could just get the version to use from there to ensure consistency

```suggestion
      - name: Read .nvmrc
        id: nvmrc      
        run: |
          echo "node-version=$(cat .nvmrc)" >> $GITHUB_OUTPUT
      - name: Use Node.js
        uses: actions/setup-node@v3
        with:
          node-version: ${{ steps.nvmrc.outputs.node-version }}
```

---

I like the idea. I will do it everywhere!